src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -51,208 +60,254 @@ end
-  @nospecialize semi # reduce precompilation time
-
-  if get(io, :compact, false)
-    show(io, semi)
-  else
-    summary_header(io, "SemidiscretizationHyperbolic")
-    summary_line(io, "#spatial dimensions", ndims(semi.equations))
-    summary_line(io, "mesh", semi.mesh)
-    summary_line(io, "equations", semi.equations |> typeof |> nameof)
-    summary_line(io, "initial condition", semi.initial_condition)
-
-    print_boundary_conditions(io, semi)
-
-    summary_line(io, "source terms", semi.source_terms)
-    summary_line(io, "solver", semi.solver |> typeof |> nameof)
-    summary_line(io, "total #DOFs", ndofs(semi))
-    summary_footer(io)
-  end
+    @nospecialize semi # reduce precompilation time
+
+    if get(io, :compact, false)
+        show(io, semi)
+    else
+        summary_header(io, "SemidiscretizationHyperbolic")
+        summary_line(io, "#spatial dimensions", ndims(semi.equations))
+        summary_line(io, "mesh", semi.mesh)
+        summary_line(io, "equations", semi.equations |> typeof |> nameof)
+        summary_line(io, "initial condition", semi.initial_condition)
+
+        print_boundary_conditions(io, semi)
+
+        summary_line(io, "source terms", semi.source_terms)
+        summary_line(io, "solver", semi.solver |> typeof |> nameof)
+        summary_line(io, "total #DOFs", ndofs(semi))
+        summary_footer(io)
+    end
 end
 
 # type alias for dispatch in printing of boundary conditions
-const SemiHypMeshBCSolver{Mesh, BoundaryConditions, Solver} =
-      SemidiscretizationHyperbolic{Mesh, Equations, InitialCondition, BoundaryConditions,
-                                   SourceTerms, Solver} where {Equations, InitialCondition, SourceTerms}
+const SemiHypMeshBCSolver{Mesh, BoundaryConditions, Solver} = SemidiscretizationHyperbolic{
+                                                                                           Mesh,
+                                                                                           Equations,
+                                                                                           InitialCondition,
+                                                                                           BoundaryConditions,
+                                                                                           SourceTerms,
+                                                                                           Solver
+                                                                                           } where {
+                                                                                                    Equations,
+                                                                                                    InitialCondition,
+                                                                                                    SourceTerms
+                                                                                                    }
```

Comment:
Something like this might work as a less ugly alternative.
```suggestion
const SemiHypMeshBCSolver{Mesh, BoundaryConditions, Solver
                          } = SemidiscretizationHyperbolic{Mesh, Equations,
                                                                                  InitialCondition, BoundaryConditions,
                                                                                  SourceTerms, Solver
                                                                                   } where {Equations, InitialCondition,
                                                                                                   SourceTerms}
```

Reply:
Yes. Also, I'd be fine if we fence off the worst offenders with a `#! format: off` region of it becomes too unbearable.

Reply:
I believe I found an acceptable version in 84fac7988.

Reply:
```julia
const SemiHypMeshBCSolver{
Mesh,
BoundaryConditions,
Solver
} = SemidiscretizationHyperbolic{Mesh,
                             Equations,
                             InitialCondition,
                             BoundaryConditions,
                             SourceTerms,
                             Solver
                             } where {Equations,
                                      InitialCondition,
                                      SourceTerms}
```
What on Earth is that?

Reply:
Changed to saner alternative in 28c44339f